infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/impl/AbstractSszMutableComposite.java

```diff
@@ -91,7 +92,15 @@ public void set(final int index, final SszChildT value) {
 
     childrenChanges.put(index, createChangeRecordByValue(immutableValue));
 
-    sizeCache = index >= sizeCache ? index + 1 : sizeCache;
+    if (!(backingImmutableData instanceof SszStableContainerBase)) {
```

Comment:
Could we extract some protected method like `calcNewSize()` and override it in `SszMutableStableContainerBaseImpl` ?
That `instanceof` check on descendant class looks a bit hacky

Reply:
Nice idea! Implemented.